flink-streaming-java/src/main/java/org/apache/flink/streaming/api/transformations/StreamTransformation.java

```diff
@@ -221,10 +232,19 @@ public int getMaxParallelism() {
 	 * @param maxParallelism Maximum parallelism for this stream transformation.
 	 */
 	public void setMaxParallelism(int maxParallelism) {
-		Preconditions.checkArgument(maxParallelism > 0
-						&& maxParallelism <= StreamGraphGenerator.UPPER_BOUND_MAX_PARALLELISM,
-				"Maximum parallelism must be between 1 and " + StreamGraphGenerator.UPPER_BOUND_MAX_PARALLELISM
-						+ ". Found: " + maxParallelism);
+		checkArgument(
+				parallelism != ExecutionConfig.PARALLELISM_DEFAULT,
+				"A maximum parallelism can only be specified with an explicitly specified " +
+						"parallelism.");
```

Comment:
Yes, it is but the problem is that the treatment of the "default parallelism" is a bit strange. (The default parallelism is the parallelism that is set from the flink config or by the user on the command line using the `-p` parameter. Maybe we have to rework that before we can fix this.

I'm starting to think that we should maybe revert the parallelism/max-parallelism changes on the release-1.2 branch and rework the whole thing properly.